### ISSUE TITLE
Add support to mount OverlayFS filesystems

### ISSUE THREAD

**Issue by piranna** [tailhook/vagga#101]
OverlayFS just allow to access to the content of several folders from one place, similar to how Plan9 bind mounts work. Since the underlaying content needs to be already available on the container filesystem there's no security issues here. All underlying filesystems are read-only, being the top one optionally read-write.

OverlayFS is one of the alternatives being used by Docker to implement its multi-layer scheme, so it could be easily used by vagga itself when several containers are being used by a command.

**Comment by tailhook** [comment]
Yes. I plan to do that. The only reason it isn't done yet, is that it requires patched kernel. Overlayfs in user namespaces (i.e. by unprivileged user) is not supported by mainline kernel. Ubuntu has a patched kernel for that.

**Comment by piranna** [comment]
Didn't know it needs a patched kernel to add support for that, I thought it would only be a matter of define capabilities or something similar, but in any case good to know that's possible :-) Knowing that, I'll mark this as done on NodeOS and do a release-candidate waiting to get this issue fixed to finish it there :-) Thanks a lot! :-D
